### ISSUE TITLE
command history is truncated after running "vim -u NONE -N"

### ISSUE THREAD

**Issue by lilydjwg** [vim/vim#1802]
I don't know how to trigger this, but since long ago, I experience command history loss sooner or later. Here's some data my backup can show:

```
$ parallel "echo -n '{.}: '; grep -c '^:' {}" ::: */.viminfo
20160828_0345/: 765
20160828_1450/: 772
20160831_2033/: 777
20160910_1204/: 780
20160914_1842/: 834
20160917_1134/: 834
20160923_2057/: 865
20160927_2255/: 875
20160930_1657/: 877
20161007_1543/: 923
20161010_2201/: 926
20161016_1111/: 936
20161016_1932/: 938
20161020_2154/: 949
20161024_2037/: 963
20161030_2119/: 973
20161102_2126/: 974
20161104_2054/: 995
20161107_1524/: 995
20161109_2133/: 1005
20161117_1543/: 1024
20161120_2109/: 1029
20161124_1253/: 1031
20161128_1747/: 1032
20161206_1800/: 1036
20161209_2059/: 1041
20161216_1315/: 1109
20161221_1217/: 1121
20161226_1519/: 1131
20161231_1100/: 1135
20170102_1045/: 1135
20170107_1535/: 1175
20170111_2203/: 1189
20170116_1319/: 1190
20170120_1138/: 1191
20170124_2237/: 1233
20170202_1821/: 1248
20170202_1911/: 1248
20170207_1252/: 89
20170213_2012/: 77
20170219_2123/: 102
20170220_2146/: 106
20170224_2207/: 106
20170306_2025/: 144
20170310_1828/: 168
20170315_1940/: 232
20170324_1456/: 246
20170330_1416/: 251
20170407_2137/: 260
20170412_1218/: 262
20170420_2015/: 297
20170427_1757/: 300
20170501_1938/: 308
20170516_1929/: 383
20170524_1225/: 391
20170531_1237/: 443
20170607_2027/: 457
20170619_1729/: 471
20170620_1121/: 485
20170620_1430/: 485
20170620_1455/: 485
20170625_2228/: 53
20170627_1331/: 56
```

I haven't tweak about viminfo for a long time, but you can see that, the lines of command history increased and then suddenly decreased to less than 100.

My 'viminfo' is `'100,:10000,<50,s10,h,!`.

Does anyone also have this issue? Any idea about how to track this down?

**Comment by lilydjwg** [comment]
Oh I think I know why! I've used `vim -u NONE -N` to reproduce an issue days ago! That command writes viminfo with the default settings.

Is this expected behaviour? (If so, please telling people to reproduce issues with `vim -u NONE -i NONE -N` instead.)

**Comment by brammool** [comment]
Glad you could figure out why it happened, it can indeed be very annoying that this happens.

Perhaps we should have a simple way to say: Use nocompatible but don't write viminfo?  Not reading viminfo would actually also help make reproduction more reliable.

**Comment by brammool** [comment]
Hmm, perhaps we can use "vim -B" as an easy way to reproduce a Bug.  It would start in nocompatible mode, not use your .vimrc, .viminfo, plugins, etc.

**Comment by chdiza** [comment]
As far as naming goes, I think `vim --clean` is preferable to `-B`.  So is `vim --bare`.  Besides being obscure, `-B` is bound to lead people to accidentally do `-b` and possibly mess up a file.

**Comment by chdiza** [comment]
Also, sometimes one needs to reproduce a "-B(ug)" without disabling `viminfo`, because the `viminfo` is part of the reproduction.

**Comment by lilydjwg** [comment]
I prefer `vim --clean` too. But can we just make `-u NOME` imply `-i NONE`?

And there is something I still don't understand: I have a long running gvim, but after `vim -u NONE -N` the command history is truncated even after that gvim saves viminfo. Shouldn't it combine the truncated result with the longer list inside its memory and produce a long list?

**Comment by chdiza** [comment]
> But can we just make -u NOME imply -i NONE?

No, that would be not just backwards incompatible, but bad.  Sometimes you want to use a viminfo but neither a vimrc nor a gvimrc.

**Comment by chrisbra** [comment]
fixed by [8.0.0716](https://github.com/vim/vim/releases/tag/v8.0.0716)
